RELEASING.md

```diff
@@ -36,20 +36,12 @@ This document explains the release strategy for artifacts in this organization,
 
 Projects create a new branch when they need to start working on 2 separate versions of the product, with the `main` branch being the furthermost release. 
 
-### OpenSearch and OpenSearch Dashboards Branching
+### OpenSearch Distribution Branching
```

Comment:
Update TOC.

Reply:
What do you think about adding a GHA to handle updating the TOC?

Reply:
Would be awesome.

Reply:
Btw, I wrote https://github.com/dblock/danger-toc a while ago to do this for the danger ecosystem, something like that would be nice here.